xonsh/ptk_shell/formatter.py

```diff
@@ -0,0 +1,45 @@
+import builtins
+import functools
+from concurrent.futures import ThreadPoolExecutor
+
+from xonsh.prompt.base import PromptFormatter, DEFAULT_PROMPT
+
+
+class ThreadedPromptFormatter(PromptFormatter):
+    def __call__(self, template=DEFAULT_PROMPT, fields=None, threaded=False):
+        """Formats a xonsh prompt template string."""
+        self.template = template  # current template
+        self.threaded = threaded
+        self.futures = {}
+        if threaded and (not hasattr(self, "thread_pool")):
+            self.thread_pool = ThreadPoolExecutor(
+                max_workers=builtins.__xonsh__.env["ASYNC_PROMPT_THREAD_WORKERS"]
+            )
+
+        prompt = super().__call__(template, fields)
+
+        if threaded:
```

Comment:
So the class name says `Threaded` but it also supports non-threaded? Shouldn't that be out of scope for this class?

Reply:
yes the same case as said above.